src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -257,133 +263,199 @@ private void enqueueNotification(TripMonitorNotification ...tripMonitorNotificat
+        if (trip.isInactive()) return true;
+
+        // calculate the appropriate timezone to use for the target time based off of the appropriate trip end location
+        ZoneId targetZoneId = trip.timezoneForTargetLocation();
+
+        // get the most recent journey state and itinerary to see when the next monitored trip is supposed to occur
+        journeyState = trip.retrieveJourneyState();
+        if (
+            journeyState.matchingItinerary == null ||
+                journeyState.matchingItinerary.endTime.before(new Date(journeyState.lastCheckedMillis))
+        ) {
+            // Either the monitored trip hasn't ever checked on the next itinerary, or the most recent itinerary has
+            // completed and the next possible one needs to be fetched in order to determine the scheduled start time of
+            // the itinerary on the next possible day the monitored trip happens
+
+            LOG.info("Calculating next itinerary for trip");
+            // calculate target time for the next trip plan request
+            // find the next possible day the trip is active by initializing the the appropriate target time. Start by
+            // checking today's date at the earliest in case the user has paused trip monitoring for a while
+            targetZonedDateTime = DateTimeUtils.nowAsZonedDateTime(targetZoneId)
+                .withHour(trip.tripTimeHour())
+                .withMinute(trip.tripTimeMinute())
+                .withSecond(0);
+
+            // if a previous journeyState target date exists, check if the previous target date was today's date. If so,
+            // immediately advance to the next day
+            if (journeyState.targetDate != null) {
+                LocalDate lastDate = DateTimeUtils.getDateFromString(
+                    journeyState.targetDate,
+                    DEFAULT_DATE_FORMAT_PATTERN
+                );
+                if (
+                    lastDate.getYear() == targetZonedDateTime.getYear() &&
+                        lastDate.getMonthValue() == targetZonedDateTime.getMonthValue() &&
+                        lastDate.getDayOfMonth() == targetZonedDateTime.getDayOfMonth()
+                ) {
+                    targetZonedDateTime = targetZonedDateTime.plusDays(1);
+                }
+            }
+
+            // calculate the next possible itinerary. If the calculation failed, skip this trip check.
+            if (!calculateNextItinerary()) return true;
+
+            // check if the matching itinerary has already ended
+            if (matchingItinerary.endTime.before(DateTimeUtils.nowAsDate())) {
+                // itinerary is done today, advance to the next day and recheck
+                targetZonedDateTime = targetZonedDateTime.plusDays(1);
+                // calculate the next possible itinerary. If the calculation failed, skip this trip check.
+                if (!calculateNextItinerary()) return true;
+            }
```

Comment:
I'm trying to understand the net effect of these multiple calls to `calculateNextItinerary`. Let's say it's Thursday and the trip has ended. On line 319, we advance the target date to Friday and call `calculateNextItinerary`. This is where it starts to break down for me... having both of these methods modify `targetZonedDateTime` makes it really hard to follow the logic. We need some kind of real-world example included in the comments. This feels a bit too spaghetti-like for my tastes and seems to be introducing a lot of embedded complexity.

Reply:
That Thursday example assumes that the `targetDate` variable is not null in the JourneyState which would not be the case the very first time that the CheckMonitoredTrip runs for a certain MonitoredTrip. That's why the check is needed to ensure that after calculating the first target date and matching itinerary that the target date is advanced to the next date that the trip is possible. I added some comments about this in 6fd00a976c063c5b46e7254a995fbbb628b90445